crates/compiler/mono/src/inc_dec.rs

```diff
@@ -889,175 +888,41 @@ fn insert_refcount_operations_binding<'a>(
-            new_let!(stmt)
-        }
-        Expr::Call(Call {
-            arguments,
-            call_type,
-        }) => match call_type {
-            // A by name call refers to a normal function call.
-            // Normal functions take all their parameters as owned, so we can mark them all as such.
-            CallType::ByName { name, .. } => {
-                // Lowlevels are wrapped in another function in order to add type signatures which help with inference.
-                // But the reference counting algorithm inserts reference counting operations in the wrapper function.
-                // But in a later stage, calls to the wrapper function were replaced by calls to the lowlevel function.
-                // Effectively removing the inserted reference counting operations.
-                // Thus to prevent that, we inline the operations here already.
-                if let LowLevelWrapperType::CanBeReplacedBy(op) =
-                    LowLevelWrapperType::from_symbol(name.name())
-                {
-                    let borrow_signature = lowlevel_borrow_signature(arena, op);
-                    let arguments_with_borrow_signature = arguments
-                        .iter()
-                        .copied()
-                        .zip(borrow_signature.iter().copied());
-                    let owned_arguments = arguments_with_borrow_signature
-                        .clone()
-                        .filter_map(|(symbol, ownership)| ownership.is_owned().then_some(symbol));
-                    let borrowed_arguments =
-                        arguments_with_borrow_signature.filter_map(|(symbol, ownership)| {
-                            ownership.is_borrowed().then_some(symbol)
-                        });
-
-                    let new_stmt = dec_borrowed!(borrowed_arguments, stmt);
-
-                    let new_let = new_let!(new_stmt);
-
-                    inc_owned!(owned_arguments, new_let)
-                } else {
-                    let new_let = new_let!(stmt);
-
-                    inc_owned!(arguments.iter().copied(), new_let)
-                }
+    macro_rules! refcount_listget {
+        ($arguments:expr) => {{
+            // TODO this index can be used to store the children of the list symbol.
+            // On the decrement of the list (if the list size is known and) the function can be specialized.
+            let [structure, _index] = match $arguments {
+                [structure, index] => Some([*structure, *index]),
+                _ => None,
             }
-            CallType::Foreign { .. } => {
-                // Foreign functions should be responsible for their own memory management.
-                // But previously they were assumed to be called with borrowed parameters, so we do the same now.
-                let new_stmt = dec_borrowed!(arguments.iter().copied(), stmt);
+            .unwrap();
```

Comment:
what is happening here? could this not be

    let structure = match $arguments {
        [structure, index] => *structure,
        _ => unreachable!(),
    };